woodwork/exceptions.py

```diff
@@ -27,3 +27,17 @@ def get_warning_message(self, saved_version_str):
                 '%s is no longer supported by this version '
                 'of woodwork. Attempting to load woodwork.DataTable ...'
                 % (saved_version_str))
+
+
+class SchemaInvalidatedWarning(UserWarning):
+    def get_warning_message(self, attr, invalid_reason):
+        return (f'Operation performed by {attr} has invalidated the Woodwork typing information:\n '
+                f'{invalid_reason}.\n '
+                'Please reinitialize Woodwork with DataFrame.ww.init')
+
+
+class CannotInitSchemaWarning(UserWarning):
+    def get_warning_message(self, attr, invalid_reason):
+        return (f'DataFrame created by {attr} is not valid for the given typing information:\n '
+                f'{invalid_reason}.\n '
+                'Please initialize Woodwork with DataFrame.ww.init')
```

Comment:
How would you feel about combining these into a single `SchemaMismatchWarning`. I see these are warning for slightly different situations, but the warnings are similar enough I wonder if we can get by with a single one?

Reply:
I think if we say that the second Warning can exist in the else of `if isinstance(result, pd.DataFrame):`, then it's fine to have the Warnings be exactly the same.

But if, in theory, you can get both warnings at once (if there's a DataFrame operation that could both create a new DataFrame and alter the original one in a way that both had the schema invalidated), it'd be super confusing to get the same Warning twice.

But also I can change it so that you pass in whatever message you want to a generic `SchemaMismatchWarning`.

Reply:
Sure makes sense. I'm thinking we consolidate these for now until we find a situation that would require them to be separated.